logstash-core/lib/logstash/codecs/base.rb

```diff
@@ -11,7 +11,7 @@ module LogStash::Codecs; class Base < LogStash::Plugin
 
   def initialize(params={})
     super
-    config_init(@params)
+    config_init(@allow_env, @params)
```

Comment:
the idea was to split it out as soon as possible. Since it is hijacking the `params`, I wanted to get it out of there as fast as possible